File: AbstractFactoryPattern/Python/FurnitureStore/factory/art_deco_furniture_factory.py
from products.chair.art_deco_chair import ArtDecoChair
from products.chair.chair import Chair
from factory.furniture_factory import FurnitureFactory
from products.sofa.art_deco_sofa import ArtDecoSofa
from products.sofa import Sofa
from products.table import ArtDecoTable
from products.table import Table


class ArtDecoFurnitureFactory(FurnitureFactory):
    def create_chair(self) -> Chair:
        return ArtDecoChair()

    def create_sofa(self) -> Sofa:
        return ArtDecoSofa()

    def create_table(self) -> Table:
        return ArtDecoTable()

File: AbstractFactory/Java/FurnitureStore/src/com/laxmena/factory/ArtDecoFurnitureFactory.java
package com.laxmena.factory;

import com.laxmena.chair.ArtDecoChair;
import com.laxmena.chair.Chair;
import com.laxmena.sofa.ArtDecoSofa;
import com.laxmena.sofa.Sofa;
import com.laxmena.table.ArtDecoTable;
import com.laxmena.table.Table;

public class ArtDecoFurnitureFactory implements FurnitureFactory {

    @Override
    public Chair createChair() {
        return new ArtDecoChair();
    }

    @Override
    public Sofa createSofa() {
        return new ArtDecoSofa();
    }

    @Override
    public Table createTable() {
        return new ArtDecoTable();
    }
}
